### PR TITLE
Improve error messages returned by TX execution

### LINKED ISSUE #2673: Transaction execution should return a more clear error
Right now, if a transaction fails, the client reports only which keys have been changed and which VPs have rejected the transaction. This is very confusing for a user as the reason why the transaction was rejected is not clear.
We should add the ability for to return a more clear error from the WASM execution back to the client to inform the user of what when wrong.

### PR DESCRIPTION
**Pull request by sug0**
## Describe your changes
Closes #2673.

Return error messages from tx execution, instead of aborting execution with no context.

## Indicate on which release or other PRs this topic is based on

#2940

Diff for review: https://github.com/anoma/namada/compare/tiago/wasm-errs-ux..tiago/tx-errs (in PR https://github.com/anoma/namada/pull/2958/files/620d3d8403794b818a03eddd391dc1153244319e..f707ee96ab8f3abdd45d1afac820a7ce53f6cd6a)

## Checklist before merging to `draft`
- [x] I have added a changelog
- [x] Git history is in acceptable state

### CODE REVIEW

File: wasm/wasm_source/src/tx_init_proposal.rs [hunk 1]
```diff
@@ -41,12 +41,13 @@ fn apply_tx(ctx: &mut Ctx, tx: Tx) -> TxResult {
                 .map_err(|err| {
                     ctx.set_commitment_sentinel();
```
Comment (grarco) [inline wasm/wasm_source/src/tx_init_proposal.rs:42]: Maybe in a future PR we could also get rid of this other sentinel just like you did for the vps

File: crates/namada/src/vm/host_env.rs [hunk 2]
```diff
@@ -2244,6 +2250,30 @@ where
...
+    buf_ptr: u64,
+    buf_len: u64,
+) -> vp_host_fns::EnvResult<()>
+where
+    MEM: VmMemory,
+    D: 'static + DB + for<'iter> DBIter<'iter>,
+    H: 'static + StorageHasher,
+    CA: WasmCacheAccess,
+{
+    // NB: ignore gas costs, as this host fn is essentially
```
[the remark this answers is not in the document]
Comment (sug0) [reply crates/namada/src/vm/host_env.rs:2265]: now charging gas in 9f8f9f38f4740829d1fc6b9d09136b019f240d8b

File: wasm/wasm_source/src/tx_init_proposal.rs (hunk 1 above)
Comment (sug0) [reply wasm/wasm_source/src/tx_init_proposal.rs:42]: https://github.com/anoma/namada/issues/2962

File: crates/namada/src/vm/wasm/run.rs [hunk 3]
```diff
@@ -38,11 +38,18 @@ const TX_ENTRYPOINT: &str = "_apply_tx";
 const VP_ENTRYPOINT: &str = "_validate_tx";
 const WASM_STACK_LIMIT: u32 = u16::MAX as u32;
 
+/// The error type returned by transactions.
+// TODO: move this to `core`, to be shared with the wasm vm,
+// and make it an `enum` of different variants
+type TxError = String;
```
Comment (tzemanovic) [inline crates/namada/src/vm/wasm/run.rs:44]: should we make an issue for this?

**Review by tzemanovic — APPROVED** [review]

File: crates/namada/src/vm/wasm/run.rs (hunk 3 above)
Comment (tzemanovic) [reply crates/namada/src/vm/wasm/run.rs:44]: #2980

**Pull request merged**